requestmanager/client.go

```diff
@@ -61,35 +61,32 @@ type inProgressRequestStatus struct {
 	inProgressErr        chan error
 	traverser            ipldutil.Traverser
 	traverserCancel      context.CancelFunc
+	lsys                 *ipld.LinkSystem
+	reconciledLoader     *reconciledloader.ReconciledLoader
 }
 
 // PeerHandler is an interface that can send requests to peers
 type PeerHandler interface {
 	AllocateAndBuildMessage(p peer.ID, blkSize uint64, buildMessageFn func(*messagequeue.Builder))
 }
 
-// AsyncLoader is an interface for loading links asynchronously, returning
-// results as new responses are processed
-type AsyncLoader interface {
-	StartRequest(graphsync.RequestID, string) error
-	ProcessResponse(ctx context.Context, responses map[graphsync.RequestID]graphsync.LinkMetadata, blks []blocks.Block)
-	AsyncLoad(p peer.ID, requestID graphsync.RequestID, link ipld.Link, linkContext ipld.LinkContext) <-chan types.AsyncLoadResult
-	CompleteResponsesFor(requestID graphsync.RequestID)
-	CleanupRequest(p peer.ID, requestID graphsync.RequestID)
+// PersistenceOptions is an interface for getting loaders by name
+type PersistenceOptions interface {
+	GetLinkSystem(name string) (ipld.LinkSystem, bool)
```

Comment:
need docs on why there are many named link systems here

Reply:
this interface is unrelated to the current PR -- it just got moved around a bit -- can we defer to a seperate ticket? This got large